chainer/variable.py

```diff
@@ -1346,8 +1356,32 @@ def backward(self, retain_grad=False, enable_double_backprop=False,
             if loss_scale is not None:
                 self.grad *= loss_scale
 
-        with chainer.using_config('enable_backprop', enable_double_backprop):
-            _backprop_to_all([self], retain_grad, loss_scale)
+        if not return_cont:
+            with chainer.using_config(
+                    'enable_backprop', enable_double_backprop):
+                _backprop_to_all([self], retain_grad, loss_scale)
+            return
+
+        ref_self = [self]
+
+        def cont():
+            if not ref_self:
+                raise RuntimeError(
+                    'the continuation Variable.backward(_return_cont=True) '
+                    'has been consumed')
+            outputs = ref_self
+            if sys.getrefcount(ref_self[0]) > 2:
```

Comment:
Is it a good way to assert it here? How about keeping the week ref to an output and checking that it is invalid after `_backprop_to_all`?